DatadogRUM/Sources/Integrations/TelemetryInterecptor.swift

```diff
@@ -0,0 +1,33 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+
+/// Intercepts telemetry events sent through message bus.
+internal struct TelemetryInterecptor: FeatureMessageReceiver {
```

Comment:
I'd say it always starts simple at the beginning. The `TelemetryReceiver` already covers multiple concerns, I didn't want to introduce another one. For that reason I've chosen composition and I'd prefer to stay with this unless this is a change request.

While it can be easily moved to `TelemetryReceiver`, adding unit tests will be way harder as it operates on much higher abstractions than the `TelemetryInterceptor` we introduce here. LTMKWDYT @maxep

Reply:
Yes, I don't have strong opinion on this.
But the typos still persist ;)

Reply:
> But the typos still persist ;)

Ops! Fixed ✅